src/main/scala/org/ua/ilm/Main.scala

```diff
@@ -0,0 +1,38 @@
+package org.ua.ilm
+
+import java.io.File
+
+import org.ua.ilm.datamining.crawlers.youtube.{Crawler, CrawlerChannels}
+import org.ua.ilm.datastore.io.config.reader.ConfigurationFileReader
+import org.ua.ilm.datastore.io.reader.IdChannelsReader
+import org.ua.ilm.datastore.io.writer.FileWriter
+
+object Main {
+  def main(args: Array[String]): Unit = {
+    if (!args.isEmpty) {
+      try {
+        val config = new ConfigurationFileReader()
+        val youTube = new Crawler(args(0)).createYT()
+
+        val idChannels = new IdChannelsReader(new File(config.paramConfig("channels"))).readFile()
+
+        val youTubeCrawlerChannels =
+          if (args.length > 2 && !args(2).isEmpty) new CrawlerChannels(youTube, idChannels, args(2))
+          else new CrawlerChannels(youTube, idChannels)
+
+        val youTubeRequestStr = youTubeCrawlerChannels.channelListRequest()
+        val writer =
+          if (!args(1).isEmpty) new FileWriter(new File(args(1)))
+          else new FileWriter(new File(config.paramConfig("json")))
+
+        writer.writeFile(youTubeRequestStr.toPrettyString)
+      }
+      catch {
+        case exception: Exception=>println("Error: "+ exception)
```

Comment:
very common exception. what about separating code blocks and ingesting them with personal try logic?